airflow/cli/commands/variable_command.py

```diff
@@ -86,7 +87,18 @@ def variables_import(args):
         except JSONDecodeError:
             raise SystemExit("Invalid variables file.")
     suc_count = fail_count = 0
+    skipped = set()
+    action_on_existing = args.action_on_existing_key
+    if action_on_existing == "fail":
+        existing_key = set(session.scalars(select(Variable.key).where(Variable.key.in_(var_json))))
+        if existing_key:
+            raise SystemExit(f"Failed. These keys: {list(sorted(existing_key))}  already exists.")
+
     for k, v in var_json.items():
+        if action_on_existing == "skip":
+            if session.scalar(select(Variable).where(Variable.key == k)):
+                skipped.add(k)
+                continue
```

Comment:
If we have `existing_key` here, we can do

```python
for k, v in var_json.items():
    if action_on_existing == "skip" and k in existing_key:
        continue
    Variable.set(...)
```

and save the query here.

Reply:
This would mean that even if the user chose to overwrite, we will still run the existing key query because the query will no longer be under an if statement

Reply:
Is that better?

Reply:
We can do

```python
if action_on_existing == "overwrite":
    existing_key = set()
else:
   ...
```

This makes the message at the very end easier as well.